### PR TITLE
Refactor/385 tidy up spawner tests

### LINKED ISSUE #385: Tidy up Spawner Tests
During work on previous issues I noticed some things I want/need to change about the Spawner/Schedule Tests.

## Things to do:
- [x] remove unnecessary pytest decorators
- [x] remove unnecessary imports
- [x] pytest ignore `SmardApi._request`
- [x] make Smard API-mock data more realistic so that trains are actually spawned
- [x] push test coverage of my code to 100 %

### PR DESCRIPTION
**Pull request by CR1337**
Fixes #385 

I pushed coverage to 100 % and fixed some stuff I noticed could be changed.

## PR checklist

- [x] Acceptance criteria fulfilled
- [x] Additional features are tested
~~- [ ] Docs (code, wiki & diagrams) updated~~
~~- [ ] Breaking changes anounced~~
~~- [ ] Added classes that inherit from `BaseModel` to `src.constants.tables`~~
- [ ] Dev-branch has been merged into local branch to resolve conflicts
- [ ] Tests and linter have passed AFTER local merge
- [ ] Another dev reviewed and approved

### CODE REVIEW

File: tests/spawner/conftest.py [hunk 1]
```diff
@@ -61,569 +61,384 @@ def random_strategy_spawn_ticks() -> list[int]:
...
+        1183600,
+        1185400,
+        1189000,
+        1191700,
+        1195300,
+        1198900,
+        1202500,
+        1206100,
+        1208800,
     ]
```
Comment (instant-sky) [inline tests/spawner/conftest.py:443]: Why do you use so many more ticks for the demand strategy than for the random strategy?

**Review by instant-sky — APPROVED** [review]
Looks good from what I understand, awesome coverage! 💯

File: tests/spawner/conftest.py (hunk 1 above)
Comment (CR1337) [reply tests/spawner/conftest.py:443]: This is historically grown :D
There is no other particular reason than this. If necessary I will decrease the amount of ticks in the future.

**Comment by CR1337** [comment]
> Looks good from what I understand, awesome coverage! 100

Thanks :)

**Pull request merged**